tests/unit/api/objectReplicationMD.js

```diff
@@ -0,0 +1,304 @@
+const assert = require('assert');
+const async = require('async');
+
+const { cleanup, DummyRequestLogger, makeAuthInfo, TaggingConfigTester } =
+    require('../helpers');
+const { metadata } = require('../../../lib/metadata/in_memory/metadata');
+const DummyRequest = require('../DummyRequest');
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const objectDelete = require('../../../lib/api/objectDelete');
```

Comment:
Ideally unit tests should not rely on other APIs. Is there a way you can think of mocking the data?